### ISSUE TITLE
Message not sending or take really long time to send

### ISSUE THREAD

**Issue by romain-pastorelli** [FossifyOrg/Messages#176]
### Checklist

- [X] I can reproduce the bug with the latest version given [here](https://github.com/FossifyOrg/Messages/releases/latest).
- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Messages/issues) or [closed](https://github.com/FossifyOrg/Messages/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Messages/discussions) or [closed](https://github.com/FossifyOrg/Messages/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one bug.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Messages/blob/master/CONTRIBUTING.md).

### Affected app version

1.0.1

### Affected Android/Custom ROM version

Lineage 20

### Affected device model

Xiaomi Mi 9t

### How did you install the app?

F-Droid / IzzyOnDroid

### Steps to reproduce the bug

1. Write a small text only message or select a file
2. Hit send

### Expected behavior

Message sending and double check marks appearing bellow

### Actual behavior

Message doesn't arrive to the receiver and "Sending..." stays under the message for hours.

### Additional information

The error will happen over any network, any time, in any condition.
Only very few times, my message is sent directly.

**Comment by romain-pastorelli** [comment]
Yeah, it's happening with any type of message (well I didn't try every file type, but images bring this error).

I get the 'Error code : 3 (Message could not be sent, etc...)' something like that, I guess.
The strange thing is that sometimes, sending message to a number N1 works perfectly, but in the same time, sending to another number N2 bring this error. And I didn't find any pattern explaining why N1 works this time and not N2, when sometimes it's reverse (N1 doesn't work and N2 may work)...

Yes, I disabled LineageOS's stock apps

**Comment by Aga-C** [comment]
@romain-pastorelli Please enable the stock messaging app, and test if sending messages works in it. We want to make sure if it's fault on our side, or you've disabled too many services.

**Comment by romain-pastorelli** [comment]
I enabled the stock app, tried to send a message to the same number which caused the error the last time and it worked perfectly. I also figured out that this number didn't even reveive my last message (sent from your app, not the stock app), even though it was marked as sent (in both apps)

**Comment by Cwpute** [comment]
@DB9-max 
https://github.com/FossifyOrg/android-smsmms
is this what you're asking for?

**Comment by naveensingh** [comment]
The android-smsmms library is [no longer used]( https://github.com/SimpleMobileTools/Simple-SMS-Messenger/pull/543) for SMS messages, it's only used for MMS.

**Comment by jahnson** [comment]
I like this app, but also experienced the non-sending of messages on Sony Xperia's under LineageOS 20 and 21.  
In one case Messages informed me that the message was not sent.  
In the other case there was no error message, but the recipient did not receive the message.  
In both cases switching back to the stock Lineage messaging app resulted in the same messages being sent to the same recipients successfully.

**Comment by fanf** [comment]
I have the same issue, and it's getting worse. It looks like having sent/received big photos started the problem (at least it seems worse now), but now it persiss and makes the app almost unusable :(

I'm on stock oneplus 9 android oxygen os 14.
Default Google message works quickly.

**Comment by naveensingh** [comment]
You might be experiencing https://github.com/FossifyOrg/Messages/issues/574

**Comment by fanf** [comment]
@naveensingh and.... You're right! Thank you and sorry for the noise
